packages/react/src/wrap.ts

```diff
@@ -0,0 +1,110 @@
+import { createElement } from 'react'
+import type { ComponentProps, ComponentType } from 'react'
+import type { PropsWithoutChildren } from './types'
+import { AsyncBoundary, Delay, ErrorBoundary, ErrorBoundaryGroup, Suspense } from '.'
+
+type WrapperItem<
+  TWrapperComponent extends
+    | typeof Suspense
+    | typeof Suspense.CSROnly
+    | typeof ErrorBoundary
+    | typeof ErrorBoundaryGroup
+    | typeof AsyncBoundary
+    | typeof AsyncBoundary.CSROnly
+    | typeof Delay,
+> = [TWrapperComponent, PropsWithoutChildren<ComponentProps<TWrapperComponent>>]
+
+type Wrapper =
+  | WrapperItem<typeof Suspense>
+  | WrapperItem<typeof Suspense.CSROnly>
+  | WrapperItem<typeof ErrorBoundary>
+  | WrapperItem<typeof ErrorBoundaryGroup>
+  | WrapperItem<typeof AsyncBoundary>
+  | WrapperItem<typeof AsyncBoundary.CSROnly>
+  | WrapperItem<typeof Delay>
+
+class WrapWithoutCSROnly {
+  constructor(private wrappers: Wrapper[]) {}
+  Suspense = (props: PropsWithoutChildren<ComponentProps<typeof Suspense>> = {}) => {
```

Comment:
Do we need to provide default parameter??

Reply:
I thought wrap.Suspense, wrap.Suspense.CSROnly, wrap.Delay don't need props object

```tsx
const Example = wrap.Suspense(/* no parameter is available*/).on(() => {
  return 'success'
})
```

Reply:
then how about optional type define?

```suggestion
  Suspense = (props?: PropsWithoutChildren<ComponentProps<typeof Suspense>>) => {
```

<img width="854" alt="image" src="https://github.com/suspensive/react/assets/57122180/1d1941d6-52a6-4cd9-8cdd-2bfc2a12a1ff">

It can prevent unnecessary variable allocation.

Reply:
Not all wrap.XXX have optional props object. optional props object is only for wrap.Suspense, wrap.Suspense.CSROnly, wrap.Delay. wrap.ErrorBoundary, wrap.ErrorBoundaryGroup require props object